src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -1786,5 +2023,639 @@ static void Main()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var y = (C2)""dog""u8;
+        var z = (C3)""cat""u8;
+    }
+}
+
+class C2
+{
+    public static implicit operator C2(Span<byte> x)
+    {
+        return new C2();
+    }
+}
+
+class C3
+{
+    public static implicit operator C3(ReadOnlySpan<byte> x)
+    {
+        return new C3();
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            // PROTOTYPE(UTF8StringLiterals) : Confirm this is the behavior we want. We are relying on user-defined conversions to convert from byte[] to span types.
+            comp.VerifyDiagnostics(
+                // (7,17): error CS0030: Cannot convert type 'byte[]' to 'C2'
+                //         var y = (C2)"dog"u8;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, @"(C2)""dog""u8").WithArguments("byte[]", "C2").WithLocation(7, 17),
+                // (8,17): error CS0030: Cannot convert type 'byte[]' to 'C3'
+                //         var z = (C3)"cat"u8;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, @"(C3)""cat""u8").WithArguments("byte[]", "C3").WithLocation(8, 17)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void UserDefinedExplicitConversions_03()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        C1 x = ""hello""u8;
+        C2 y = ""dog""u8;
+        C3 z = ""cat""u8;
+    }
+}
+
+class C1
+{
+    public static explicit operator C1(byte[] x)
+    {
+        return new C1();
+    }
+}
+
+class C2
+{
+    public static explicit operator C2(Span<byte> x)
+    {
+        return new C2();
+    }
+}
+
+class C3
+{
+    public static explicit operator C3(ReadOnlySpan<byte> x)
+    {
+        return new C3();
+    }
+}";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            comp.VerifyDiagnostics(
+                // (7,16): error CS0266: Cannot implicitly convert type 'byte[]' to 'C1'. An explicit conversion exists (are you missing a cast?)
+                //         C1 x = "hello"u8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, @"""hello""u8").WithArguments("byte[]", "C1").WithLocation(7, 16),
+                // (8,16): error CS0029: Cannot implicitly convert type 'byte[]' to 'C2'
+                //         C2 y = "dog"u8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""dog""u8").WithArguments("byte[]", "C2").WithLocation(8, 16),
+                // (9,16): error CS0029: Cannot implicitly convert type 'byte[]' to 'C3'
+                //         C3 z = "cat"u8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""cat""u8").WithArguments("byte[]", "C3").WithLocation(9, 16)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
```

Comment:
>[ConditionalFact(typeof(CoreClrOnly))] [](http://example.com/codeflow?start=8&length=38)

`[Fact]`

Reply:
HelpersSource refers to types that are not available on desktop. I am fairly comfortable to test this scenario only on CoreClr.